WEB/Src/DependencyCollector/Shared.Tests/Implementation/AzureSdkDiagnosticListenerTest.cs

```diff
@@ -0,0 +1,147 @@
+    using Microsoft.ApplicationInsights.Web.TestFramework;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class AzureSdkDiagnosticListenerTest
+    {
+        private TelemetryConfiguration configuration;
+        private List<ITelemetry> sentItems;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+            this.configuration = new TelemetryConfiguration();
+            this.sentItems = new List<ITelemetry>();
+            this.configuration.TelemetryChannel = new StubTelemetryChannel { OnSend = item => this.sentItems.Add(item), EndpointAddress = "https://dc.services.visualstudio.com/v2/track" };
+            this.configuration.InstrumentationKey = Guid.NewGuid().ToString();
+        }
+
+        [TestCleanup]
+        public void CleanUp()
+        {
+            while (Activity.Current != null)
+            {
+                Activity.Current.Stop();
+            }
+        }
+
+        [TestMethod]
+        public void AzureClientSpansAreCollected()
+        {
+            using (var listener = new DiagnosticListener("Azure.SomeClient"))
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+
+                Activity sendActivity = null;
+                Activity parentActivity = new Activity("parent").AddBaggage("k1", "v1").Start();
+                parentActivity.TraceStateString = "state=some";
+                var telemetry = this.TrackOperation<DependencyTelemetry>(
+                    listener,
+                    "Azure.SomeClient.Send",
+                    null,
+                    () => sendActivity = Activity.Current);
+
+                Assert.IsNotNull(telemetry);
+                Assert.AreEqual("Send", telemetry.Name);
+                Assert.IsTrue(telemetry.Success.Value);
+
+                Assert.AreEqual($"|{sendActivity.TraceId.ToHexString()}.{sendActivity.ParentSpanId.ToHexString()}.", telemetry.Context.Operation.ParentId);
+                Assert.AreEqual(sendActivity.TraceId.ToHexString(), telemetry.Context.Operation.Id);
+                Assert.AreEqual($"|{sendActivity.TraceId.ToHexString()}.{sendActivity.SpanId.ToHexString()}.", telemetry.Id);
+
+                Assert.AreEqual("v1", telemetry.Properties["k1"]);
+                Assert.AreEqual("eventhubname.servicebus.windows.net", telemetry.Properties["property"]);
+
+                Assert.IsTrue(telemetry.Properties.TryGetValue("tracestate", out var tracestate));
+                Assert.AreEqual("state=some", tracestate);
+            }
+        }
+
+        [TestMethod]
+        public void AzureClientSpansAreMarkedAsFailed()
+        {
+            using (var listener = new DiagnosticListener("Azure.SomeClient"))
+            using (var module = new DependencyTrackingTelemetryModule())
```

Comment:
Can you add one test with EnableAzureSdkTelemetryListener = false?

Reply:
Done